src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -40,20 +38,38 @@ function isErrorNonFatal(error) {
   return error.message === OVERWRITE_REJECTED || error.message === SAVE_DUPLICATE_REJECTED;
 }
 
-export function SavedObjectProvider(esAdmin, kbnIndex, Promise, Private, Notifier, confirmModalPromise, indexPatterns) {
+export function SavedObjectProvider(esAdmin, kbnIndex, Promise, Private, Notifier, confirmModalPromise, indexPatterns, savedObjectsClient) {
 
-  const DocSource = Private(AdminDocSourceProvider);
   const SearchSource = Private(SearchSourceProvider);
   const mappingSetup = Private(MappingSetupProvider);
 
+  let savedObjectsQueue = [];
+
+  const throttledSavedObjectsBatch = _.throttle(() => {
+    const queue = _.clone(savedObjectsQueue);
+    savedObjectsQueue = [];
+
+    savedObjectsClient.bulkGet(queue).then(({ savedObjects }) => {
+      queue.forEach((item) => {
+        const foundObject = savedObjects.find(savedObject => {
+          return savedObject.id === item.id;
+        });
+
+        if (!foundObject) {
+          item.reject(new SavedObjectNotFound(item.type, item.id));
+        }
+
+        item.resolve(_.cloneDeep(foundObject));
+      });
+    });
+  }, 2000, { leading: false });
```

Comment:
thoughts on changing the 2000 to a constant to make it a little bit more clear? `COLLECTION_TIME` or similar.